### PR TITLE
Fix AO statute mapping where section contains a letter

### PR DESCRIPTION
**Pull request by lbeaufort**
## Summary (required)

- Addresses https://github.com/fecgov/openFEC/issues/2803

_Fix bug where REGEX isn't accounting for characters (IE "2 USC 441b" is being parsed as "2 USC 441") which is causing lookup mistakes. 1980-01 is an example. Add tests for handling these characters properly_

### Before
~2 U.S.C. §437f == 52 U.S.C. §30105~
~2 U.S.C. §441b == 2 U.S.C. §441~

### After
2 U.S.C. §437f == 52 U.S.C. §30108
2 U.S.C. §441b == 52 U.S.C. §30118
2 U.S.C. §441a-1 == 52 U.S.C. §30117

## How to test the changes locally

- Get Elasticsearch up and running
- get the latest S3 credentials from `dev`
- run `./manage.py initialize_current_legal_docs` which recreates the docs index and loads current AO's. Quit the process when it gets to MURs - not needed.
- Check http://localhost:5000/v1/legal/search/?type=advisory_opinions&ao_no=1980-01&ao_category=F and the statutory citations should be 52 30108 and 52 30118 (vs. [dev](https://fec-dev-api.app.cloud.gov/v1/legal/search/?type=advisory_opinions&ao_no=1980-01&ao_category=F) which still shows 2 441)

## Impacted areas of the application
List general components of the application that this PR will affect:

-  https://www.fec.gov/data/legal/advisory-opinions/

### CODE REVIEW

File: webservices/legal_docs/advisory_opinions.py [hunk 1]
```diff
@@ -60,7 +67,7 @@
 """
 
 STATUTE_CITATION_REGEX = re.compile(
-    r"(?P<title>\d+)\s+U\.?S\.?C\.?\s+§*\s*(?P<section>\d+).*\.?")
+    r"(?P<title>\d+)\s+U\.?S\.?C\.?\s+§*\s*(?P<section>\d+\d+[-0-9a-z]*).*\.?")
```
Comment (vrajmohan) [inline webservices/legal_docs/advisory_opinions.py]: This has the undesired consequence of requiring that section numbers be at least _2_ digits followed by 0 or more digit-or-lowercase-alphabet-or-dash.

e.g. try parsing `52 U.S.C. 3`.

File: tests/test_advisory_opinions.py [hunk 2]
```diff
@@ -35,13 +35,14 @@ def test_parse_ao_citations(text, ao_nos, expected):
...
 @pytest.mark.parametrize("text,expected", [
-    ("2 U.S.C. 432h", set([(52, 30102)])),
-    ("52 U.S.C. 30116a", set([(52, 30116)])),
-    ("2 U.S.C. 441b, 441c, 441e", set([(2, 441)])),
-    (" 2 U.S.C. §437f", set([(52, 30105)])),
-    ("52 U.S.C. § 30101", set([(52, 30101)])),
-    (" 2 USC §437f **test with no . in USC**", set([(52, 30105)])),
-    ("52 U.S.C. § 30101 **newline needed to ensure two entries**,\n 52 USC. § 30101 other words", set([(52, 30101)])),
+    ("2 U.S.C. 432", set([(52, '30102')])),
+    ("52 U.S.C. 30116(a", set([(52, '30116')])),
```
Comment (vrajmohan) [inline tests/test_advisory_opinions.py]: Now that we allow alphabets and hyphens in the section, we should add a test like:
```
("52 U.S.C. 30116a", set([(52, '30116a')])),
```

Comment (lbeaufort) [reply tests/test_advisory_opinions.py]: Thank you! I had added a test for 2 U.S.C. §441a-1 == 52 U.S.C. §30117 but I somehow lost it when re-working the (very stale) branch. I'll add it back.

Comment (lbeaufort) [reply tests/test_advisory_opinions.py]: Oh nevermind, I see it now. I'll add a "52 U.S.C. 30116a === 52 U.S.C. 30116a" test too.

File: webservices/legal_docs/advisory_opinions.py (hunk 1 above)
Comment (lbeaufort) [reply webservices/legal_docs/advisory_opinions.py]: Thank you, I've actually spent some time trying to understand REGEX better and I have a long way to go. It looks like the 2nd `\d+` is the problem, so it should be `(?P<title>\d+)\s+U\.?S\.?C\.?\s+§*\s*(?P<section>\d+[-0-9a-z]*).*\.?`. Do you agree?

Comment (vrajmohan) [reply webservices/legal_docs/advisory_opinions.py]: `(?P<section>\d+[-0-9a-z]*)` matches "one or more digits that may _optionally_ be followed by 0 or more digit-or-lowercase-alphabet-or-dash and would also match "123a-4".

`(?P<section>\d+[-a-z]*)` matches "one or more digits that may _optionally_ be followed by 0 or more lowercase-alphabet-or-dash and would match "123a" and "123a-b" but not "123a-4".

Not sure which one is better, but my hunch is the latter.

Comment (lbeaufort) [reply webservices/legal_docs/advisory_opinions.py]: Thanks @vrajmohan! I think we want to catch  2 USC 441a-1 so I'll test the first option.  http://uscode.house.gov/editorialreclassification/t52/Reclassifications_Title_52.html

**Comment by lbeaufort** [comment]
@vrajmohan and @pkfec I've addressed Vraj's earlier comments and this is ready for your review. Thanks!

**Pull request merged**